socorro/unittest/external/es/test_supersearch.py

```diff
@@ -1781,3 +1783,109 @@ def test_get_with_too_many(self):
             self.api.get,
             _results_number=1001,
         )
+
+    @minimum_es_version('1.0')
+    @requests_mock.Mocker(real_http=True)
+    def test_get_with_failing_shards(self, mock_requests):
+        # Test with one failing shard.
+        es_results = '''{
+    "hits": {
+        "hits": [],
+        "total": 0,
+        "max_score": null
+    },
+    "timed_out": false,
+    "took": 194,
+    "_shards": {
+        "successful": 9,
+        "failed": 1,
+        "total": 10,
+        "failures": [
+            {
+                "status": 500,
+                "index": "fake_index",
+                "reason": "foo bar gone bad",
+                "shard": 3
+            }
+        ]
+    }
+}'''
+
+        mock_requests.get(
+            'http://localhost:9200/{}/crash_reports/_search'.format(
+                self.config.elasticsearch.elasticsearch_index
+            ),
+            text=es_results
+        )
+
+        res = self.api.get()
+        ok_('errors' in res)
+
+        errors_exp = [
+            {
+                'type': 'shards',
+                'index': 'fake_index',
+                'shards_count': 1,
+            }
+        ]
+        eq_(res['errors'], errors_exp)
+
+        # Test with several failures.
+        es_results = '''{
+    "hits": {
+        "hits": [],
+        "total": 0,
+        "max_score": null
+    },
+    "timed_out": false,
+    "took": 194,
+    "_shards": {
+        "successful": 9,
+        "failed": 3,
+        "total": 10,
+        "failures": [
+            {
+                "status": 500,
+                "index": "fake_index",
+                "reason": "foo bar gone bad",
+                "shard": 2
+            },
+            {
+                "status": 500,
+                "index": "fake_index",
+                "reason": "foo bar gone bad",
+                "shard": 3
+            },
+            {
+                "status": 500,
+                "index": "other_index",
+                "reason": "foo bar gone bad",
+                "shard": 1
+            }
+        ]
+    }
+}'''
```

Comment:
I prefer that we write python dicts instead. Then you can do
```python
mock_requests.get(
    ...,
    text=json.dumps(es_results)
)
```
When it's python dicts you get syntax highlighting, you can put commas at the last item of a list, you can easily inject interesting values an you immediately get a `SyntaxError` instead of some later cryptic hard-to-debug JSON parse error.